### PR TITLE
Remove ref in output

### PR DESCRIPTION
**Pull request by wild-endeavor**
# TL;DR
This removes the `ref` property in the default NamedTuple object that gets used when a user want to name outputs.  This is done so that some can use an output named "ref" if they wanted to.  Previously it was getting co-opted by this property.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
The property iterates through the list of outputs, and returns the first `NodeOutput` it sees, and the node gets extracted from that.  Instead of doing this, just store the Node object in the Case object, upon construction.

## Tracking Issue
https://github.com/flyteorg/flyte/issues/3923

### CODE REVIEW

**Review by eapolinario — DISMISSED** [review]

**Review by pingsutw — COMMENTED** [review]
nit: linting error

**Review by pingsutw — APPROVED** [review]

**Comment by honnix** [comment]
Thanks for fixing this. I'm wondering whether it makes sense to have reserved prefix/suffix for this type of internal properties? A simple `_` prefix might do it.
